ext/tracer_tag_propagation/tracer_tag_propagation.c

```diff
@@ -45,8 +46,13 @@ void ddtrace_add_tracer_tags_from_header(zend_string *headerstr, zend_array *roo
                 strncmp(ZSTR_VAL(tag_name), "_dd.p.", sizeof("_dd.p.") - 1) == 0) {
                 zval zv;
                 ZVAL_STRINGL(&zv, valuestart, header - valuestart);
-                zend_hash_update(root_meta, tag_name, &zv);
-                zend_hash_add_empty_element(propagated_tags, tag_name);
+                if (zend_string_equals_literal(tag_name, DD_P_TS_KEY)) {
+                    ddtrace_trace_source_set_from_hexadecimal(Z_STR_P(&zv), root_meta);
+                    zval_ptr_dtor(&zv);
+                } else {
```

Comment:
This is now redundant and can be removed.

Reply:
Why? The incoming `_dd.p.ts` needs to still be properly parsed and populated accordingly when `trace source` is ASM

Reply:
Why do we need to parse it? The else branch takes care of populating. The only thing this essentially does is sanitization, which we don't really need here.